cmd/internal/find/parser/expression/parser.go

```diff
@@ -183,15 +183,22 @@ func (parser *Parser) Parse(tokens []string) (predicate.Predicate, []string, err
 	}
 	if _, ok := parser.stack.Peek().(*BinaryOp); ok {
 		// This codepath is possible via something like "p1 -and" or "p1 -and <unknown_token>"
-		if err != nil {
-			// We have "p1 -and <unknown_token>"
-			return nil, nil, err
+		if err == nil {
+			// We have "p1 -and"
+			return nil, nil, fmt.Errorf(
+				"%v: no expression after %v",
+				parser.stack.mostRecentOpToken,
+				parser.stack.mostRecentOpToken,
+			)
 		}
-		return nil, nil, fmt.Errorf(
-			"%v: no expression after %v",
-			parser.stack.mostRecentOpToken,
-			parser.stack.mostRecentOpToken,
-		)
+		// We have "p1 -and <unknown_token>". Pop the binary op off the stack and include
```

Comment:
This ends up being a kind of two-look-ahead parser. Interesting set of special-casing for binary ops. I worry about other edge-cases we might be missing, but don't have any specifically in mind.